### PR TITLE
Revert "Update formatter to not write newlines if content is empty (#11193)"

### LINKED ISSUE #11340: Get-Help does not display whole example body
# Steps to reproduce

```powershell
(Get-Help Out-Null).Examples
```

# Expected behavior
**PowerShell 5.1**

```none
> (Get-Help Out-Null).Examples

------------------- Example 1: Delete output -------------------

PS C:\> Get-ChildItem | Out-Null

This command gets items in the current location/directory, but its output is not passed through the pipeline nor
displayed at the command line. This is useful for hiding output that you do not need.
```

# Actual behavior
**PowerShell 7.0**

```none
> (Get-Help Out-Null).Examples

------------------- Example 1: Delete output -------------------

Get-ChildItem | Out-Null
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-preview.6
PSEdition                      Core
GitCommitId                    7.0.0-preview.6-59-gca202da6437f5c9ed334725469f6d8f1cb7e5a2c
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

This reverts commit 51d2523f5fc3953e53902941659f607465a62053. (#11193)

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/11340

It seems we filter out a null too late.
We need to have more info about a scenario we tried to fix in #11193.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/FormatAndOutput/common/ComplexWriter.cs [hunk 1]
```diff
@@ -49,24 +49,14 @@ internal void WriteObject(List<FormatValue> formatValueList)
             // we always start with no indentation
```
Comment (PoshChan) [inline src/System.Management.Automation/FormatAndOutput/common/ComplexWriter.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
XmlCommand DRT basic functionality Tests.Import-Clixml StopProcessing should succeed
```powershell
Expected 'Stopped', but got Completed.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\XMLCommand.Tests.ps1: line 77
77: 		$ps.InvocationStateInfo.State | Should -Be "Stopped"
```

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
Should we add a test for the Get-Help scenario?

**Comment by iSazonov** [comment]
> Should we add a test for the Get-Help scenario?

The scenario is a side effect. If we change Get-Help we can lost the new test. I don't know how create more common and reliable test.

**Comment by SteveL-MSFT** [comment]
@iSazonov not suggesting we change `Get-Help` itself, but a test for `(Get-Help Out-Null).Examples`

**Comment by iSazonov** [comment]
I will pull the test in follow PR (revert PR is not right place).

**Comment by daxian-dbw** [comment]
@iSazonov @SteveL-MSFT Since #11193 was reverted, is there an issue opened to track the original problem that PR was trying to fix?

**Comment by iSazonov** [comment]
@daxian-dbw Original issue was discovered by @SteveL-MSFT but not registered so we haven't tracking issue. I do not know whether it is important for @SteveL-MSFT.

**Pull request merged**